src/ctl.c

```diff
@@ -2276,6 +2276,17 @@ arena_i_decay_ms_ctl_impl(tsd_t *tsd, const size_t *mib, size_t miblen,
 			ret = EINVAL;
 			goto label_return;
 		}
+		if (arena_is_huge(arena_ind) && *(ssize_t *)newp > 0) {
+			/*
+			 * By default the huge arena purges eagerly.  If it is
```

Comment:
I'm sort of inclined to say that the user shouldn't be allowed to change purging settings for the huge arena; WDYT?

Reply:
I was thinking about that too. However if there's some valid case where huge allocations are freed and reused often (or rather bad application code), maybe we should have some mechanism in place, which might be easier to fix the usage?

Reply:
Oh I agree with you. I'm thinking this as a hack (i.e. not documented). I also think the huge arena thing should not be user visible. Right now we don't even provide a way to figure out which arena is for huge (other than doing a huge allocation and check).

I was thinking this case for, when someone comes in and says they have a weird and hard-to-change workload, do we have a fix for them  -- I also agree that we should not encourage such solutions; but given that we have the huge arena enabled by default, I feel we should be more prepared.